### PR TITLE
feat(core): dialog auto labeling

### LINKED ISSUE #6052: core(dialog): dialog cannot be labelled automatically (from the template)
#### Is this a bug, enhancement, or feature request?

Enhancement, bug.

#### Briefly describe your proposal.

As dialogs might be created from the code it would be great to have the ability to automatically label the dialog as it's not possible to specify `ariaLabelledBy` in the config at the moment of creation.

Example:

We have `TableP13DialogComponent` component which creates different dialogs, for example `P13SortingDialogComponent`. 

As we want `aria-labelledby` attribute for every `P13SortingDialogComponent` instance (and any other dialogs) we should create a counter in the `TableP13DialogComponent` component, update the counter when any sort (and others) dialogs are created, generate and pass the ID (looks like `sort-dialog-${id}`) to the dialog config and to the `P13SortingDialogComponent` to reference created ID with the title in the template.

We should repeat these steps for any similar cases.

#### Which versions of Angular and Fundamental Library for Angular are affected? (If this is a feature request, use current version.)

Latest.

#### If this is a bug, please provide steps for reproducing it.

.

#### Please provide relevant source code if applicable.

.

#### Is there anything else we should know?

[Example](https://material.angular.io/components/dialog/overview) from Angular Material.

### PR DESCRIPTION
**Pull request by platon-rov**
#### Please provide a link to the associated issue.

Closes #6052.

#### Please provide a brief summary of this pull request.

Dialog auto labeling feature using the directive.

#### Please check whether the PR fulfills the following requirements

- [X] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [X] tests for the changes that have been done
- [X] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [X] Run npm run build-pack-library and test in external application

Documentation checklist:
- [n/a] update `README.md`
- [n/a] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [X] Documentation Examples
- [X] Stackblitz works for all examples

### CODE REVIEW

File: apps/docs/src/app/core/component-docs/dialog/dialog-docs.component.html [hunk 1]
```diff
@@ -103,6 +103,21 @@
     <fd-dialog-configuration-example></fd-dialog-configuration-example>
 </component-example>
 <code-example [exampleFiles]="customConfiguration"></code-example>
+
+<separator></separator>
+
+<fd-docs-section-title id="autoLabel" componentName="dialog">
```
Comment (kavya-b) [inline apps/docs/src/app/core/component-docs/dialog/dialog-docs.component.html:109]:
```suggestion
<fd-docs-section-title id="auto-label" componentName="dialog">
```
In [this PR](https://github.com/SAP/fundamental-ngx/pull/6034#issuecomment-883341410), I am converting ids used in `*.docs.component.html` files to kebab-case for consistency, so could you make this change for `autoLabel` as well?

File: libs/core/src/lib/dialog/dialog.component.ts [hunk 2]
```diff
@@ -98,6 +99,14 @@ export class DialogComponent extends DialogBase implements OnInit, OnChanges, Af
         }
     }
 
+    /** @hidden */
+    @ContentChild(DialogTitleDirective)
+    set dialogTitle(title: DialogTitleDirective) {
+        if (title) {
+            this.dialogConfig.ariaLabelledBy = title.titleId;
```
Comment (kavya-b) [inline libs/core/src/lib/dialog/dialog.component.ts:106]: wouldn't this overwrite any values that `this.dialogConfig.ariaLabelledBy` previously had? Or are we expecting the user not to use both auto labelling and `ariaLabelledBy` together?

File: libs/core/src/lib/dialog/dialog.module.ts [hunk 3]
```diff
@@ -24,6 +24,7 @@ import { DialogDefaultComponent } from './dialog-default/dialog-default.componen
 import { DialogCloseButtonComponent } from './dialog-close-button/dialog-close-button.component';
 import { DialogDecisiveButtonDirective } from './directives/dialog-decisive-button.directive';
 import { TitleModule } from '@fundamental-ngx/core/title';
+import { DialogTitleDirective } from './directives/dialog-title.directive';
```
Comment (kavya-b) [inline libs/core/src/lib/dialog/dialog.module.ts:27]: please organize the ordering of the imports.

**Comment by kavya-b** [comment]
On opening the dialog, the focus doesn't go anywhere in the dialog, but in most other examples focus goes to the button; is this expected?

**Comment by platon-rov** [comment]
> On opening the dialog, the focus doesn't go anywhere in the dialog, but in most other examples focus goes to the button; is this expected?

It's expected, but to have consistent examples I've added an initial focus directive to the buttons.

**Review by kavya-b — APPROVED** [review]
LGTM!

**Review by Smiranin — APPROVED** [review]

**Review by olshansky — APPROVED** [review]

**Review by dimamarksman — APPROVED** [review]

**Pull request merged**